src/v1beta/os_login_service_client.ts

```diff
@@ -0,0 +1,898 @@
+   *     {@link https://developers.google.com/identity/protocols/application-default-credentials Application Default Credentials},
+   *     your project ID will be detected automatically.
+   * @param {function} [options.promise] - Custom promise module to use instead
+   *     of native Promises.
+   * @param {string} [options.apiEndpoint] - The domain name of the
+   *     API remote host.
+   */
+
+  constructor(opts?: ClientOptions) {
+    // Ensure that options include the service address and port.
+    const staticMembers = this.constructor as typeof OsLoginServiceClient;
+    const servicePath =
+      opts && opts.servicePath
+        ? opts.servicePath
+        : opts && opts.apiEndpoint
+        ? opts.apiEndpoint
+        : staticMembers.servicePath;
+    const port = opts && opts.port ? opts.port : staticMembers.port;
+
+    if (!opts) {
+      opts = {servicePath, port};
+    }
+    opts.servicePath = opts.servicePath || servicePath;
+    opts.port = opts.port || port;
+    opts.clientConfig = opts.clientConfig || {};
+
+    const isBrowser = typeof window !== 'undefined';
+    if (isBrowser) {
+      opts.fallback = true;
+    }
+    // If we are in browser, we are already using fallback because of the
+    // "browser" field in package.json.
+    // But if we were explicitly requested to use fallback, let's do it now.
+    const gaxModule = !isBrowser && opts.fallback ? gax.fallback : gax;
+
+    // Create a `gaxGrpc` object, with any grpc-specific options
+    // sent to the client.
+    opts.scopes = (this.constructor as typeof OsLoginServiceClient).scopes;
+    const gaxGrpc = new gaxModule.GrpcClient(opts);
+
+    // Save the auth object to the client, for use by other methods.
+    this.auth = gaxGrpc.auth as gax.GoogleAuth;
+
+    // Determine the client header string.
+    const clientHeader = [`gax/${gaxModule.version}`, `gapic/${version}`];
+    if (typeof process !== 'undefined' && 'versions' in process) {
+      clientHeader.push(`gl-node/${process.versions.node}`);
+    } else {
+      clientHeader.push(`gl-web/${gaxModule.version}`);
+    }
+    if (!opts.fallback) {
+      clientHeader.push(`grpc/${gaxGrpc.grpcVersion}`);
+    }
+    if (opts.libName && opts.libVersion) {
+      clientHeader.push(`${opts.libName}/${opts.libVersion}`);
+    }
+    // Load the applicable protos.
+    // For Node.js, pass the path to JSON proto file.
+    // For browsers, pass the JSON content.
+
+    const nodejsProtoPath = path.join(
+      __dirname,
+      '..',
+      '..',
+      'protos',
+      'protos.json'
+    );
+    const protos = gaxGrpc.loadProto(
+      opts.fallback ? require('../../protos/protos.json') : nodejsProtoPath
+    );
+
+    // This API contains "path templates"; forward-slash-separated
+    // identifiers to uniquely identify resources within the API.
+    // Create useful helper objects for these.
+    this._pathTemplates = {
+      posixAccountPathTemplate: new gaxModule.PathTemplate(
```

Comment:
`fingerprintPathTemplate` is lost :(  Need to figure out what's going on with it.

Reply:
in gapic yaml `collections`, it does have resource defined as 
```
  - name_pattern: users/{user}/**sshPublicKeys**/{fingerprint}
    entity_name: fingerprint
```
while in the protos, this resource is not directly used, but referenced as 
```
    (google.api.resource_reference) = {
      type: "oslogin.googleapis.com/**SshPublicKey**"
    }
 OR
   string parent = 1 [(google.api.resource_reference) = {
                       child_type: "oslogin.googleapis.com/**SshPublicKey**"
                     }];
```
so we have `sshPublicKeyPathTemplate` but no `fingerprintPathTemplate`

Also one question is that as stated in [API conversion sheet](https://docs.google.com/spreadsheets/d/1Hyn_uPH1bzCK-9JAeUz_bj7feUg41h5hcuPzZRqUj8Q/edit#gid=0) this API use undefined resource `posixAccount` in ts file, we have `posixAccountPathTemplate` which is referenced to `project`:
```
      posixAccountPathTemplate: new gaxModule.PathTemplate(
        'users/{user}/projects/{project}'
      ),
```
Not sure if this is qualified, but all tests passed until now

Reply:
Okay, thank you for investigation. I feel we might want to release it with the changed resource name, but it must be a major release (a breaking change). Let's merge it as `feat!` and not release yet (until we have a strategy about Node.js v8 deprecation).